### PR TITLE
New block header

### PR DESCRIPTION
**Pull request by EyeOfPython**

### CODE REVIEW

File: src/primitives/block.cpp [hunk 1]
```diff
@@ -7,6 +7,7 @@
 
 #include <hash.h>
 #include <tinyformat.h>
+#include <util/strencodings.h>
 
 BlockHash CBlockHeader::GetHash() const {
     return BlockHash(SerializeHash(*this));
```
Comment (schancel) [inline src/primitives/block.cpp:13]: This will need to be updated at some point for the hashing format you wanted.

File: src/primitives/block.h [hunk 2]
```diff
@@ -22,38 +28,91 @@
...
+
+    SERIALIZE_METHODS(CBlockMetadataField, obj) {
+        READWRITE(obj.nFieldId);
+        READWRITE(obj.vData);
+    }
 };
 
 class CBlock : public CBlockHeader {
 public:
+    std::vector<CBlockMetadataField> vMetadata;
```
Comment (schancel) [inline src/primitives/block.h:115]: Shouldn't this be in the header? Or are we doing it as a subtree out of the gate?

**Review by schancel — CHANGES_REQUESTED** [review]
This all makes sense. Is the goal here to just change the header in this commit, and then the hashing algorithm in another commit? I think that would make sense.

**Comment by EyeOfPython** [comment]
I think having the hashing in this commit also makes more sense. I want the tests to pass, but then I'd have to redo all the PoW etc.

We should probably split the metadata part out though. Maybe always enforce it as the hash of '\0' for this PR.

**Comment by schancel** [comment]
Closing, this has been superceeded.

**Pull request closed without merging**